airflow/providers/google/cloud/triggers/bigquery.py

```diff
@@ -113,6 +116,14 @@ async def run(self) -> AsyncIterator[TriggerEvent]:  # type: ignore[override]
                         self.poll_interval,
                     )
                     await asyncio.sleep(self.poll_interval)
+        except asyncio.CancelledError:
```

Comment:
Triggers also seemed to be cancelled when triggerer job restarts which can cause the job to be cancelled.

Reply:
There is a separate issue for this one: https://github.com/apache/airflow/issues/36090 This change can be done as part of a separate PR which changes airflow core code to handle `on_kill`.